.github/workflows/main.yml

```diff
@@ -92,67 +85,68 @@ jobs:
         run: |
           echo "::set-output name=${{ matrix.os_short }}::https://storage.googleapis.com/taqueria-artifacts/${{ steps.upload-file.outputs.uploaded }}"
 
-  build-and-publish-npm-packages:
-    runs-on: ubuntu-latest
-    permissions:
-      packages: write
-      contents: read
-    strategy:
-      matrix:
-        include:
-          # - dir: taqueria-plugin-flextesa
-          - dir: taqueria-plugin-ligo
-          - dir: taqueria-plugin-smartpy
-          - dir: taqueria-plugin-taquito
-          # - dir: taqueria-plugin-teztnets
-          - dir: taqueria-sdk
-    defaults:
-      run:
-        working-directory: ${{ matrix.dir }}
-    env: 
-      NODE_AUTH_TOKEN: ${{ secrets.NPMJS_PAT }}
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
-        with: 
-          registry-url: ${{ env.NPM_REGISTRY }}
-          scope: "${{ env.SCOPE }}"
-
-      - name: Check tag suffix
-        id: check-tag-suffix
-        if: startsWith(github.ref, 'refs/tags/v') && (contains(github.ref, '-rc') || contains(github.ref, '-beta') || contains(github.ref, '-test'))
-        run: |
-          echo "::set-output name=npm-tag::next"
-
-      - name: Install any dependencies
-        run: npm ci
+  publish-protocol-to-npm:
+    needs: build-binaries
+    uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@alex-npm-pre-build-ordering
+    with:
+      dir: taqueria-protocol
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPMJS_PAT }}
+
+  publish-sdk-to-npm:
+    needs: publish-protocol-to-npm
+    uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@alex-npm-pre-build-ordering
+    with:
+      dir: taqueria-sdk
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPMJS_PAT }}
+
+  publish-ligo-plugin-to-npm:
+    needs: publish-sdk-to-npm
+    uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@alex-npm-pre-build-ordering
+    with:
+      dir: taqueria-plugin-ligo
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPMJS_PAT }}
+  
+  publish-smartpy-plugin-to-npm:
+    needs: publish-sdk-to-npm
+    uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@alex-npm-pre-build-ordering
+    with:
+      dir: taqueria-plugin-smartpy
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPMJS_PAT }}
+
+  publish-taquito-plugin-to-npm:
+    needs: publish-sdk-to-npm
+    uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@alex-npm-pre-build-ordering
+    with:
+      dir: taqueria-plugin-taquito
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPMJS_PAT }}
+
+  publish-flextesa-plugin-to-npm:
+    needs:
+      - publish-ligo-plugin-to-npm
+      - publish-smartpy-plugin-to-npm
+      - publish-taquito-plugin-to-npm
+    uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@alex-npm-pre-build-ordering
```

Comment:
The branch name should be removed for these

Reply:
@GImbrailo you're right but they need to be changed to `main`. From the docs I could find (https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow) this is the only way I could get them to call correctly. I tried to get it to call from the other workflow file but kept getting an error that made no sense to me. I think this is another peculiarity with github actions

Reply:
You could have also split this PR and have a separate one creating the reusable workflow so that you could merge them in order.

Anyways you can also remove the branch name right before merging to `main`